### ISSUE TITLE
Test questions

### ISSUE THREAD

**Issue by zhenghw111** [mit-han-lab/bevfusion#143]
When I execute the "torchpack dist-run -np 8 python tools/test.py configs/nuscenes/det/transfusion/secfpn/camera+lidar/swint_v0p075/convfuser.yaml pretrained/bevfusion-det.pth --eval bbox" command,it stops at the following position:
![image](https://user-images.githubusercontent.com/108120679/191894180-8b593a5e-c196-4ca0-ac67-405a1cb23d2d.png)

**Comment by kentang-mit** [comment]
Would you mind trying out single GPU evaluation (i.e. `-np 1`) or adding the `-v` flag to your command? Besides, there are two on-going PRs working on docker support for BEVFusion. I would also recommend you to have a try at them:

[PR](https://github.com/mit-han-lab/bevfusion/pull/144) from @bentherien and [PR](https://github.com/mit-han-lab/bevfusion/pull/145) from me.

**Comment by zhenghw111** [comment]
Thank! This problem has been solved.
Can you tell me the training command?

**Comment by kentang-mit** [comment]
@zhenghw111, the training code is under internal review and quality check, we will keep you posted on the progress of code release.

**Comment by zhenghw111** [comment]
OK, thank you!

**Comment by kkangshen** [comment]
作者你好，请问模型架构中fuser层的作用是什么？

**Comment by kkangshen** [comment]
@kentang-mit  作者你好，请问模型架构中fuser层的作用是什么？

**Comment by kkangshen** [comment]
@kentang-mit @zhenghw111 请问在bevfusion.py中，camera得到的future的维度为[1,80,128,128]，lidar得到的future的维度为[1，256，180，180]，他们的维度最后两维不相等，这是为什么呢？
![image](https://user-images.githubusercontent.com/39619373/192184604-c5c65873-d538-41c4-a797-62eaef20a47a.png)

**Comment by kentang-mit** [comment]
Hi @kkangshen,

I will first translate your questions into English such that other readers might understand the context better.

1. What's the function of the `fuser` in BEVFusion?

This module is used to combine the information from camera and LiDAR sensors. In our experiments, we confirm that it is sufficient to use only one convolution layer (followed by BN/ReLU).

2. Mismatched dimensionalities: the output of the camera encoder is in [1, 80, 128, 128] and the output of the LiDAR encoder is [1, 256, 180, 180]. 

I think I kept output resolution of both branches the same. For detection it is [180, 180], and for segmentation it is [128, 128]. 

Best,
Haotian

**Comment by kkangshen** [comment]
Hi @kentang-mit 
Thanks for your answer, but there are still some problems here, if the two dimensions are not equal the output of the camera encoder is in [1, 80, 128, 128] and the output of the LiDAR encoder is [1, 256, 180, 180 ], there will be an error when **torch.cat(inputs, dim=1)** how should this problem be solved.
![image](https://user-images.githubusercontent.com/39619373/192516171-3e6ac91e-5625-4ce1-a552-3c5efb5af95e.png)

**Comment by kentang-mit** [comment]
Hi @kkangshen,

I would suggest you to use the same output resolutions for both branches. 

If you really want to have different resolutions for camera and LiDAR branches, I would suggest you to have a look at [this class](https://github.com/mit-han-lab/bevfusion/blob/2bf96604feab90edd18591a43bee1b9c41c26002/mmdet3d/models/heads/segm/vanilla.py#L47) in our segmentation head implementation, which actually changes the resolution of the output BEV feature map to a target size. You can use a similar resolution to align the resolutions of both branches first and then concatenate them together for further processing.

Best,
Haotian

**Comment by kkangshen** [comment]
s

**Comment by kkangshen** [comment]
Hi @kentang-mit 
Thank you very much for replying，I have the following questions。
1. I run the test.py file torchpack dist-run -np 8 python tools/test.py configs/nuscenes/det/transfusion/secfpn/camera+lidar/swint_v0p075/convfuser.yaml pretrained/bevfusion- det.pth --eval bbox,
When running, it is found that the dimension obtained by extract_camera_features is different from the dimension obtained by extract_lidar_features. If I want to make the two feature dimensions the same, how should I modify it.
2. I want to do the detection task, do I need to convert the features obtained by extract_camera_features into [1, 80, 180, 180]
3. When I passed the debug script, I found that when the checkpoint = load_checkpoint(model, args.checkpoint, map_location="cpu") was executed, the dx, nx, and bx parameters in the configuration file were changed. I don't know if this caused the extract_camera_features The feature becomes [1, 80, 128, 128]

**Comment by kentang-mit** [comment]
That sounds pretty interesting to me @kkangshen. I can confirm that your observation in 3 is reasonable because `dx`, `nx` and `bx` are defined as model parameters, so loading the checkpoint might override these values in the configuration file. But I actually have tested these pretrained models we released and I did not run into this issue. Would you mind downloading the latest checkpoints again (because we updated them very recently) and make sure that the correct checkpoint is loaded (for example, please do not load the segmentation checkpoint if you want to do detection)?

**Comment by zhenghw111** [comment]
@kkangshen 请问在pycharm上调试时是如何启动torchpack库的？我现在无法调试...

**Comment by kentang-mit** [comment]
The problem related to PyCharm debugging has been resolved in another issue. I'm closing this one since it is resolved.

**Comment by kkangshen** [comment]
> @kkangshen 请问在pycharm上调试时是如何启动torchpack库的？我现在无法调试...
你将dist.init()注释掉 ，使用单个gpu就可以

**Comment by lilkeker** [comment]
@kkangshen I also want to debug on pycharm now. I can run the program in the terminal, but the debug parameter is passed in and an error is reported.
![BE(9K XCJ%401MW4}SOSGSM](https://user-images.githubusercontent.com/81810465/194760490-5a5d7ef1-caba-432e-a7f4-3ad95db82f76.png)
When debugging, I commented out dist.init() as you said, and set distributed to False,
![Y C`(BX$B3UJFQ$J8L} TD3](https://user-images.githubusercontent.com/81810465/194760624-62cda82c-1de4-4721-8c29-faa7b4530b61.png)
I think my dataloader is fine, because the terminal input command can run the program,do you know why？please
